frappe/integrations/webhooks.py

```diff
@@ -0,0 +1,27 @@
+# Copyright (c) 2015, Frappe Technologies Pvt. Ltd. and Contributors
+# MIT License. See license.txt
+
+from __future__ import unicode_literals
+import frappe, requests
+from frappe import _
+
+# Doc Events Webhook
+def doc_event_webhook(doc, method=None, *args, **kwargs):
+	headers = {}
+	filters = {
+		"webhook_doctype": doc.get("doctype"),
+		"webhook_docevent": method
+	}
+	webhooks = frappe.get_all("Webhook", filters=filters)
+	webhook = frappe.get_doc("Webhook", webhooks[0].get("name")) if webhooks and len(webhooks) > 0 else None
+	if webhook:
+		if webhook.webhook_headers:
+			for h in webhook.webhook_headers:
+				if h.get("key") and h.get("value"):
+					headers[h.get("key")] = h.get("value")
```

Comment:
you should be able to pull value from the `doc` too. That should be apart of standard use case. Where are you using webhooks?

Maybe introduce a "parameter_type" as `Static` or `Document Property`

Reply:
Use case : 
For cloud/SaaS users to add a webhook.
When the DocType and DocEvent mentioned in Webhook is triggered POST request will be made to the request_url and doc will be passed.
e.g. User deletes a Note, infroms the event by POST request to request_url, if webhook for Note and doc_event is added.
like https://developer.github.com/webhooks

I didn't understand the following part,

> you should be able to pull value from the doc too

I need the saved `Webhook` for the given doc, not the doc itself.

I tried get_doc, but it checks for every event triggered. For example, on Save of existing Note, on_update and on_change is also triggered and it throws error because there is no Webhook for every event triggered.
So I used get_all to first get list or empty list. this doesn't throw error.

Headers for the request are optional.
if required by any app then header like `Authorization` or `X-API-KEY` can be manually added.

`paramater_type` in headers child table? I didn't get it.